src/components/Identity/createComponent.js

```diff
@@ -0,0 +1,179 @@
+import createCustomerIds from "./customerIds/createCustomerIds";
+import { defer } from "../../utils";
+import { EXPERIENCE_CLOUD_ID } from "./constants/cookieNames";
+import migration from "./migration";
+import createEvent from "../DataCollector/createEvent";
+
+const addIdsContext = (payload, ecid) => {
+  payload.addIdentity(EXPERIENCE_CLOUD_ID, {
+    id: ecid
+  });
+};
+
+export default (idSyncs, manualIdSyncs, cookieJar, config, logger, network) => {
+  let optIn;
+  let deferredForEcid;
+  let lifecycle;
+  let customerIds;
+  let alreadyQueriedForIdSyncs = false;
+  const { migrateIds, imsOrgId } = config;
+
+  const createPayload = event => {
+    const payload = network.createPayload();
+    payload.addEvent(event);
+    return payload;
+  };
+
+  const readEcid = () => {
+    let ecid = cookieJar.get(EXPERIENCE_CLOUD_ID);
+    if (!ecid && migrateIds) {
+      ecid = migration(imsOrgId).readEcidFromAmcvCookie();
+      if (ecid) {
+        cookieJar.set(EXPERIENCE_CLOUD_ID, ecid);
+      }
+    }
+    return ecid;
+  };
+
+  const makeServerCall = payload => {
+    return lifecycle.onBeforeDataCollection({ payload }).then(() => {
+      return network.sendRequest(payload, payload.expectsResponse);
+    });
+  };
+
+  const createEcidEvent = () => {
```

Comment:
I'm a little worried about this and I'm not sure we thought the whole thing through. Creating those server calls to fetch one value are weird, because the response contains much more than the ECID. Let's talk this over with Rares.

Reply:
It is for the `getEcid` command.
What is the better way to return an `Ecid` if there was no `viewstart`  event.

Reply:
Removed the network call.
We will add it if there is an endpoint just returning ID